packages/dds/task-manager/src/test/taskManager.spec.ts

```diff
@@ -280,6 +281,99 @@ describe("TaskManager", () => {
                 assert.ok(taskManager2.assigned(taskId), "Task manager 2 should be assigned");
             });
         });
+
+        describe("Completing tasks", () => {
+            it("Can complete a task", async () => {
+                const taskId = "taskId";
+                const volunteerTaskP = taskManager1.volunteerForTask(taskId);
+                containerRuntimeFactory.processAllMessages();
+                await volunteerTaskP;
+                assert.ok(taskManager1.assigned(taskId), "Should be assigned");
+
+                taskManager1.complete(taskId);
+                containerRuntimeFactory.processAllMessages();
```

Comment:
Re: my comment above about event timing -- I think that before calling `processAllMessages()` here, `taskManager1.assigned(taskId) === false` but the "completed" event won't fire until after calling `processAllMessages()`.